### PR TITLE
partial of the new queue service

### PR DESCRIPTION
**Pull request by thorwolpert**
*Issue #:* /bcgov/entity#16405

*Description of changes:*
Partial add of the new alternate queue. Backed off some changes and the simple cloudevent for the older version of python.
This takes care of Step1 - add minimal publish to sbc-pay

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Comment by thorwolpert** [comment]
Code Qual: I'm leaving the broad Exceptions to be replaced by the Q lib when we can move to a newer version of Python

File: pay-api/requirements/repo-libraries.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 -e git+https://github.com/bcgov/sbc-common-components.git#egg=sbc-common-components&subdirectory=python
+git+https://github.com/daxiom/simple-cloudevent.py.git@0.0.2
```
Comment (seeker25) [inline pay-api/requirements/repo-libraries.txt:2]: bcgov repo?

File: pay-api/src/pay_api/services/gcp_queue_publisher.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    # get authenticated publisher
+    try:
+        service_account_info = json.loads(base64.b64decode(gcp_auth_key).decode('utf-8'))
+        credentials = jwt.Credentials.from_service_account_info(
+            service_account_info, audience=audience
+        )
+        credentials_pub = credentials.with_claims(audience=publisher_audience)
+        publisher = pubsub_v1.PublisherClient(credentials=credentials_pub)
+    except Exception as error:  # noqa: B902
+        raise Exception('Unable to create a connection', error) from error  # pylint: disable=W0719
```
Comment (seeker25) [inline pay-api/src/pay_api/services/gcp_queue_publisher.py:47]: Might want to create our own exceptions for these?

File: pay-api/requirements.txt [hunk 3]
```diff
@@ -73,3 +77,4 @@ typing_extensions==4.5.0
 urllib3==1.26.15
 zipp==3.15.0
 -e git+https://github.com/bcgov/sbc-common-components.git#egg=sbc-common-components&subdirectory=python
+git+https://github.com/daxiom/simple-cloudevent.py.git
```
Comment (seeker25) [inline pay-api/requirements.txt:80]: bcgov repo?

File: pay-api/requirements/repo-libraries.txt (hunk 1 above)
Comment (thorwolpert) [reply pay-api/requirements/repo-libraries.txt:2]: Following the same pattern as Patrick, with the local repo to start

File: pay-api/src/pay_api/services/gcp_queue_publisher.py (hunk 2 above)
Comment (thorwolpert) [reply pay-api/src/pay_api/services/gcp_queue_publisher.py:47]: When I move this to the common library, I will do that. I bumped into some changes between py3.8 and ^3.9 that caused some issues.

File: pay-api/requirements/repo-libraries.txt (hunk 1 above)
Comment (seeker25) [reply pay-api/requirements/repo-libraries.txt:2]: Ok, as long as this gets cleaned up later

**Review by seeker25 — APPROVED** [review]

File: pay-api/src/pay_api/services/gcp_queue_publisher.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    ce.data = payload
+
+    _send_to_queue(to_queue_message(ce))
+
+
+def _send_to_queue(payload: bytes):
+    """Send payload to the queue."""
+    if not ((gcp_auth_key := current_app.config.get('gcp_auth_key')) and
+            (audience := current_app.config.get('audience')) and
+            (topic_name := current_app.config.get('topic_name')) and
```
Comment (kialj876) [inline pay-api/src/pay_api/services/gcp_queue_publisher.py:33]: I think these config references are case sensitive aren't they?

**Review by kialj876 — APPROVED** [review]
Looks good to me, but might want to verify my comment above if you haven't already

File: pay-api/src/pay_api/services/gcp_queue_publisher.py (hunk 4 above)
Comment (thorwolpert) [reply pay-api/src/pay_api/services/gcp_queue_publisher.py:33]: I'll re-run my tests to verify that my global search/replace didn't screw me up.

**Pull request merged**